spec/log_weasel/middleware_spec.rb

```diff
@@ -69,6 +69,40 @@
         end
       end
 
+      context "when the log weasel id is included in the params" do
+        let(:env) do
+          Rack::MockRequest.env_for("something", params: { logweasel_id: 'foo' })
+        end
+
+        context "with the environment variable enabled" do
+          it "sets LogWeasel::Transation.id to the parameter value" do
+            allow(ENV).to receive(:fetch).with('LOG_WEASEL_FROM_PARAMS', nil).and_return(true)
+
+            expect(StitchFix::LogWeasel::Transaction).to receive(:id=).with("foo")
+
+            StitchFix::LogWeasel::Middleware.new(app).call(env)
+          end
+        end
+      end
+
+      context "when the log weasel id is not included in the params" do
+        let(:env) do
+          Rack::MockRequest.env_for("something", params: { something_else: 'foo' })
+        end
+
+        context "with the environment variable enabled" do
+          it "does not set the log weasel id from the params" do
+            allow(ENV).to receive(:fetch).with('LOG_WEASEL_FROM_PARAMS', nil).and_return(true)
+
+            env['HTTP_X_REQUEST_ID'] = 'bar'
```

Comment:
This was more straightforward than trying to mock or stub out all potential calls on `env`